libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1365,7 +1606,44 @@ impl From<cln::ListpeersPeersChannels> for Channel {
     }
 }
 
-/// Conversion for a closed channel
+fn convert_to_send_pay_route(
+    route: PaymentPath,
+    to_pay_msat: u64,
+    final_cltv_delta: u64,
+) -> (Vec<pb::cln::SendpayRoute>, u64) {
+    let mut sendpay_route = vec![];
+    let mut to_forward = to_pay_msat;
+    let mut cltv_delay = 0;
+    let hops_arr = route.edges.as_slice();
+
+    for i in 1..hops_arr.len() + 1 {
```

Comment:
A reverse loop here would be a little easier to follow (since `i` is not used)

Reply:
We need to iterate over the reversed route and to be able also to update the previous iterated hop (as forwarding hop) with fees and cltv delta.
Any idea how to simplify it?

Reply:
OK, I fixed to use enumerate() which provides also the index.